### ISSUE TITLE
Natvis: are multi-dimensional arrays supported in gdb natvis?

### ISSUE THREAD

**Issue by elbrandt** [microsoft/MIEngine#980]
Hello,
I have a natvis to visualize a 2D array that works perfectly in VisualStudio 2019.
The same natvis file on VSCode 1.43.0 in Ubunutu, using gdb is not so happy.

Can you tell me if multi dimensional array capability is supposed to work using MIMode: gdb?

I'll be glad to work up a full minimal-working-example if this is supposed to work, but for now, here's the Type definition from my natvis that works on VStudio, but not on VSCode/gdb

``` <!-- ARRAY<T, VECTOR<T, 2>> -->
  <Type Name="PhysBAM::ARRAYS_ND_BASE&lt;PhysBAM::VECTOR&lt;*,2&gt;&gt;">
    <DisplayString>{{ [({domain.min_corner.x},{domain.min_corner.y}),({domain.max_corner.x},{domain.max_corner.y})] }}</DisplayString>
    <Expand>
      <ArrayItems>
        <Direction>Forward</Direction>
        <Rank>2</Rank>
        <Size>((int*)&amp;counts)[$i]</Size>
        <ValuePointer>($T1*)base_pointer</ValuePointer>
      </ArrayItems>
    </Expand>
  </Type>
```

Thanks in advance!
-Eric.

View in VSCode (not so good):
![image](https://user-images.githubusercontent.com/36109794/77333947-295bf800-6cf2-11ea-9d90-8c08b1edff04.png)

View in Visual Studio (very nice!):
![image](https://user-images.githubusercontent.com/36109794/77332593-3b3c9b80-6cf0-11ea-9542-00ae42fb2c48.png)

**Comment by pieandcakes** [comment]
@paulmaybee Is this something that is supported in our MIEngine's natvis implementation?

**Comment by paulmaybee** [comment]
@pieandcakes Multi-dimensional array are not currently supported.

**Comment by elbrandt** [comment]
Thanks for letting me know...saves me the trouble (for the time being) of trying to debug something that isn't meant to work.

Do you have any idea of where this support ranks in terms of priority, difficulty, mi-support, etc? Is this something that is 'likely to happen' or 'wishful thinking'?

**Comment by Trass3r** [comment]
I guess more the latter.
The are more pressing issues with natvis than this.

**Comment by WardenGnaw** [comment]
Fixed with #1346

**Comment by sean-mcmanus** [comment]
The fix is available with 1.13.1: https://github.com/microsoft/vscode-cpptools/releases/tag/v1.13.1
